### PR TITLE
[cli] Add test for gatsby cache

### PR DESCRIPTION
**Pull request by styfle**
This PR adds a E2E CLI test to ensure that the Gatsby example deploys correctly and that the second deployment has the proper cached directories.

### CODE REVIEW

File: packages/now-cli/test/integration.js [hunk 1]
```diff
@@ -297,6 +299,30 @@ test('deploy using only now.json with `redirects` defined', async t => {
...
+      cwd,
+      stdio: 'inherit',
+      reject: true,
+    });
+
+    t.true(true);
+  }
+
+  await tryDeploy(directory);
+  await sleep(5000);
```
Comment (TooTallNate) [inline packages/now-cli/test/integration.js]: Why 5 seconds?

File: packages/now-cli/test/integration.js [hunk 2]
```diff
@@ -297,6 +299,30 @@ test('deploy using only now.json with `redirects` defined', async t => {
...
+  pkg.scripts.build = `ls -lA && ls .cache && ls public && ${pkg.scripts.build}`;
+  await writeFile(packageJsonPath, JSON.stringify(pkg));
+  await tryDeploy(directory);
+  await writeFile(packageJsonPath, packageJsonOriginal);
```
Comment (TooTallNate) [inline packages/now-cli/test/integration.js]: Restoring the original package.json should happen in a `finally` block.

File: packages/now-cli/test/integration.js [hunk 3]
```diff
@@ -297,6 +299,30 @@ test('deploy using only now.json with `redirects` defined', async t => {
...
+    });
+
+    t.true(true);
+  }
+
+  await tryDeploy(directory);
+  await sleep(5000);
+  pkg.scripts.build = `ls -lA && ls .cache && ls public && ${pkg.scripts.build}`;
+  await writeFile(packageJsonPath, JSON.stringify(pkg));
+  await tryDeploy(directory);
```
Comment (TooTallNate) [inline packages/now-cli/test/integration.js]: Are we missing some asserts after the deploy?

File: packages/now-cli/test/integration.js [hunk 4]
```diff
@@ -297,6 +299,30 @@ test('deploy using only now.json with `redirects` defined', async t => {
...
+      reject: true,
+    });
+
+    t.true(true);
+  }
+
+  await tryDeploy(directory);
+  await sleep(5000);
+  pkg.scripts.build = `ls -lA && ls .cache && ls public && ${pkg.scripts.build}`;
+  await writeFile(packageJsonPath, JSON.stringify(pkg));
```
Comment (TooTallNate) [inline packages/now-cli/test/integration.js]: _Could_ use `writeJson()` from fs-extra, but not a big deal either way.

File: packages/now-cli/test/integration.js (hunk 3 above)
Comment (styfle) [reply packages/now-cli/test/integration.js]: No because previously, this second deploy would fail. The assertion is that tryDeploy() has `reject: true`.

File: packages/now-cli/test/integration.js (hunk 1 above)
Comment (styfle) [reply packages/now-cli/test/integration.js]: It seemed like the deployment was returning a 0 status code prior to it being completely done but perhaps that was a different issue. Let me try again without the delay.

File: packages/now-cli/test/integration.js (hunk 2 above)
Comment (styfle) [reply packages/now-cli/test/integration.js]: Done 👍

File: packages/now-cli/test/integration.js (hunk 1 above)
Comment (styfle) [reply packages/now-cli/test/integration.js]: Ok so I updated to 60 seconds because S3 is terrible and we have to wait for the cache to propagate (yes the cache response is cached 🤦‍♂️ )

File: packages/now-cli/test/integration.js (hunk 4 above)
Comment (styfle) [reply packages/now-cli/test/integration.js]: I want to delete `fs-extra` in the future so don't want to add new code using it 😄

File: packages/now-cli/test/integration.js [hunk 5]
```diff
@@ -2934,3 +2936,36 @@ test('`vc --debug project ls` should output the projects listing', async t => {
...
+    await execa(binaryPath, [...defaultArgs, '--public', '--confirm'], {
+      cwd,
+      stdio: 'inherit',
+      reject: true,
+    });
+
+    t.true(true);
+  }
+
+  // Deploy once too populate the cache
```
Comment (AndyBitz) [inline packages/now-cli/test/integration.js]:
```suggestion
  // Deploy once to populate the cache
```

**Review by AndyBitz — DISMISSED** [review]

**Review by TooTallNate — APPROVED** [review]
